src/v/datalake/datalake_manager.cc

```diff
@@ -282,35 +281,33 @@ ss::future<> datalake_manager::start() {
     co_await _backlog_controller->start();
 
     /*
-     * Start the global disk space usage monitor loop
+     * Initialize the reservation pool and start the global disk space usage
+     * monitor loop
      */
     const ss::shard_id disk_space_monitor_core = 0;
     if (ss::this_shard_id() == disk_space_monitor_core) {
+        _disk_space_available.signal(
+          config::shard_local_cfg().datalake_scratch_space_size_bytes());
         ssx::spawn_with_gate(_gate, [this] { return disk_space_monitor(); });
     }
+}
+
+size_t datalake_manager::disk_space_soft_limit() {
+    return config::shard_local_cfg()
+      .datalake_scratch_space_soft_limit_size_bytes();
+}
 
-    _disk_usage_interval.watch([this] { _disk_space_monitor_sem.signal(); });
+bool datalake_manager::disk_space_soft_limit_exceeded() {
+    return _disk_space_available.current() < disk_space_soft_limit();
 }
 
 ss::future<> datalake_manager::disk_space_monitor() {
     while (!_gate.is_closed()) {
-        const auto interval = _disk_usage_interval();
-        try {
-            co_await _disk_space_monitor_sem.wait(
-              interval, std::max(_disk_space_monitor_sem.current(), size_t(1)));
-        } catch (const ss::semaphore_timed_out& ex) {
-            std::ignore = ex;
-            // drop through to perform work
-        }
-
-        if (_disk_usage_interval() != interval) {
-            // configuration change
-            continue;
-        }
-
-        if (!config::shard_local_cfg().datalake_disk_space_monitor_enable()) {
-            continue;
-        }
+        co_await _disk_space_monitor_cv.wait([this] {
+            return disk_space_soft_limit_exceeded()
```

Comment:
nit: probably not important at all but i'd do `config::shard_local_cfg(). datalake_disk_space_monitor_enable() && disk_space_soft_limit_exceeded()` to short-circuit the check and never go into `disk_space_soft_limit_exceeded` if the feature is disabled

Reply:
i see you responded to other comments. did you get a chance to read this one? if so, resolving the comment or giving it a thumbs up is helpful for managing feedback.